HW3.py

```diff
@@ -0,0 +1,88 @@
+# ЗАДАЧА-1
+# Написать свой декоратор который будет проверять остаток от деления числа 100 на результат работы функции ниже.
+# Если остаток от деления = 0, вывести сообщение "We are OK!», иначе «Bad news guys, we got {}» остаток от деления.
+
+from functools import wraps
+
+
+# def check_decorator(func):
+#     @wraps(func)
+#     def inner(arg):
+#         res = 100 % func(arg)
+#         if res == 0:
+#             print("We are OK!")
+#         else:
+#             print(f"Bad news guys, we got {res}")
+#     return inner
+#
+#
+# @check_decorator
+# def sqr(arg):
+#     """The function squares the arg"""
+#     return arg**2
+#
+#
+# sqr(3)
+
+
+# ЗАДАЧА-2
+# Написать декоратор который будет выполнять предпроверку типа аргумента который передается в вашу функцию.
+# Если это int, тогда выполнить функцию и вывести результат, если это str(),
+# тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+
+
+# def check_type_decorator(func):
+#     @wraps(func)
+#     def inner(a):
+#         arg = func(a)
+#         value_type = type(arg)
+#         if value_type == int:
+#             print(arg)
+#         elif value_type == str:
```

Comment:
'if isinstance(arg, int)', 'if isinstance(arg, str)'  is more preferable for data type validation